cub/cub/device/device_reduce.cuh

```diff
@@ -102,6 +170,85 @@ struct unzip_and_write_arg_extremum_op
 //! @endrst
 struct DeviceReduce
 {
+private:
+  // TODO(gevtushenko): dispatch to atomic reduce once merged
+  template <typename TuningEnvT,
+            typename InputIteratorT,
+            typename OutputIteratorT,
+            typename ReductionOpT,
+            typename T,
+            typename NumItemsT,
+            ::cuda::execution::determinism::__determinism_t Determinism>
+  CUB_RUNTIME_FUNCTION static cudaError_t reduce_impl(
+    void* d_temp_storage,
+    size_t& temp_storage_bytes,
+    InputIteratorT d_in,
+    OutputIteratorT d_out,
+    NumItemsT num_items,
+    ReductionOpT reduction_op,
+    T init,
+    ::cuda::execution::determinism::__determinism_holder_t<Determinism>,
+    cudaStream_t stream)
+  {
+    using offset_t        = detail::choose_offset_t<NumItemsT>;
+    using accum_t         = ::cuda::std::__accumulator_t<ReductionOpT, detail::it_value_t<InputIteratorT>, T>;
+    using transform_t     = ::cuda::std::identity;
+    using reduce_tuning_t = ::cuda::std::execution::
+      __query_result_or_t<TuningEnvT, detail::reduce::get_tuning_query_t, detail::reduce::default_tuning>;
+    using policy_t = typename reduce_tuning_t::template fn<accum_t, offset_t, ReductionOpT>;
```

Comment:
_as I am working on TransformReduce env based APIs I ran across the following_

Here we let `policy_t` know of the `RedutionOpT`.
In line `#L194` we let `accum_t` also know of the `ReductionOpT`. 
We pass both to `dispatch_t`.

Couldn't we just query/deduce the `ReductionOpT` in the `using fn = ...` definitions to avoid passing it twice and make the code less error prone? 

food for thought ...